benches/wasm-kernel/src/rev_complement.rs

```diff
@@ -0,0 +1,155 @@
+//! Adapted version from benchmarks game. In particular
+//! rayon is removed.
+//!
+// https://benchmarksgame-team.pages.debian.net/benchmarksgame/program/revcomp-rust-3.html
+
+
+use std::io::BufRead;
+use std::{cmp, io};
+use std::mem::replace;
+
+/// Lookup table to find the complement of a single FASTA code.
+fn build_table() -> [u8; 256] {
+	let mut table = [0; 256];
+	for (i, x) in table.iter_mut().enumerate() {
+		*x = match i as u8 as char {
+			'A' | 'a' => 'T',
+			'C' | 'c' => 'G',
+			'G' | 'g' => 'C',
+			'T' | 't' => 'A',
+			'U' | 'u' => 'A',
+			'M' | 'm' => 'K',
+			'R' | 'r' => 'Y',
+			'W' | 'w' => 'W',
+			'S' | 's' => 'S',
+			'Y' | 'y' => 'R',
+			'K' | 'k' => 'M',
+			'V' | 'v' => 'B',
+			'H' | 'h' => 'D',
+			'D' | 'd' => 'H',
+			'B' | 'b' => 'V',
+			'N' | 'n' => 'N',
+			i => i,
+		} as u8;
+	}
+	table
+}
+
+/// Utilities for splitting chunks off of slices.
+trait SplitOff {
+	fn split_off_left(&mut self, n: usize) -> Self;
+	fn split_off_right(&mut self, n: usize) -> Self;
+}
+impl<'a, T> SplitOff for &'a mut [T] {
+	/// Split the left `n` items from self and return them as a separate slice.
+	fn split_off_left(&mut self, n: usize) -> Self {
+		let n = cmp::min(self.len(), n);
+		let data = replace(self, &mut []);
+		let (left, data) = data.split_at_mut(n);
+		*self = data;
+		left
+	}
+	/// Split the right `n` items from self and return them as a separate slice.
+	fn split_off_right(&mut self, n: usize) -> Self {
+		let len = self.len();
+		let n = cmp::min(len, n);
+		let data = replace(self, &mut []);
+		let (data, right) = data.split_at_mut(len - n);
+		*self = data;
+		right
+	}
+}
+
+/// Length of a normal line including the terminating \n.
+const LINE_LEN: usize = 61;
+
+/// Compute the reverse complement for two contiguous chunks without line breaks.
+fn reverse_chunks(left: &mut [u8], right: &mut [u8], table: &[u8; 256]) {
+	for (x, y) in left.iter_mut().zip(right.iter_mut().rev()) {
+		*y = table[replace(x, table[*y as usize]) as usize];
+	}
+}
+
+/// Compute the reverse complement on chunks from opposite ends of a sequence.
+///
+/// `left` must start at the beginning of a line. If there are an odd number of
+/// bytes, `right` will initially be 1 byte longer than `left`; otherwise they
+/// will have equal lengths.
+fn reverse_complement_left_right(mut left: &mut [u8],
```

Comment:
You shouldn't have indentation that relies on the length of identifiers if you're using tabs for indentation. Just `rustfmt` and this will be fixed.